.github/workflows/on-push.yaml

```diff
@@ -0,0 +1,254 @@
+name: On Push
+on: [push]
+
+concurrency:
+  # limit concurrency of entire workflow runs for a specific branch
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  NODE_VERSION: 22.1.0
+  BUF_VERSION: 1.42.0
+  PYTHON_VERSION: 3.13
+  CARGO_INCREMENTAL: 0
+  CARGO_PROFILE_DEV_STRIP: "debuginfo"
+
+jobs:
+
+  # Runs various lints and checks for the project, including Rustfmt, Clippy,
+  # Protobuf lints, and pnpm lints.
+  lint:
+    name: Run Lints
+    runs-on: ubuntu-24.04
+    steps:
+      - uses: rui314/setup-mold@v1
+        with:
+          make-default: true
```

Comment:
Why the mold linker? Is it just faster?

Reply:
Yup, generally speaking a lot faster than ld